### PR TITLE
REST API for core

### PR DESCRIPTION
**Pull request by tdudzik**
REST API for core.

Possibilities:
- create list of books: POST http://localhost:8090
- retrieve all books: GET http://localhost:8090

I've wrote also a mapper (DTO -> Entity, Entity -> DTO) using the mapper from http://modelmapper.org/.

### CODE REVIEW

File: Common/src/main/java/javapocalypse/common/dto/BookDTO.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import lombok.*;
+
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class BookDTO {
+
+    @Getter
+    @Setter
+    private long id;
```
Comment (caroo7) [inline Common/src/main/java/javapocalypse/common/dto/BookDTO.java:13]: Why you need 'id' in DTO?

File: Core/src/main/java/javapocalypse/core/api/BooksController.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        this.bookMapper = bookMapper;
+    }
+
+    @RequestMapping(path = "", method = RequestMethod.GET, produces = "application/json")
+    @ResponseBody
+    public List<BookDTO> listBooks() {
+        return bookMapper.mapToDTOs(bookRepository.findAll());
+    }
+
+    @RequestMapping(path = "", method = RequestMethod.POST, consumes = "application/json", produces = "application/json")
```
Comment (caroo7) [inline Core/src/main/java/javapocalypse/core/api/BooksController.java:30]: :+1:

**Comment by caroo7** [comment]
:+1:

File: Common/src/main/java/javapocalypse/common/dto/BookDTO.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                .add("title", title)
+                .add("authors", authors)
+                .add("description", description)
+                .add("oldPrice", oldPrice)
+                .add("newPrice", newPrice)
+                .add("tag", tag)
+                .toString();
+    }
+
+}
```
Comment (egolesor) [inline Common/src/main/java/javapocalypse/common/dto/BookDTO.java:57]: Where you need all this setters and getters ?

File: Core/src/main/java/javapocalypse/core/Application.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package javapocalypse.core;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.ComponentScan;
+
+@ComponentScan
```
Comment (egolesor) [inline Core/src/main/java/javapocalypse/core/Application.java:9]: I think ComponentScan is not needed and the variable ctx albo is not used so you can just write SpringApplication.run(Application.class). Maybe you have a reason for that ?

File: Core/src/main/java/javapocalypse/core/api/BooksController.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+import javapocalypse.core.model.entities.Book;
+import javapocalypse.core.model.repositories.BookRepository;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.web.bind.annotation.*;
+
+import javax.inject.Inject;
+import java.util.List;
+
+@RestController
+@EnableAutoConfiguration
```
Comment (egolesor) [inline Core/src/main/java/javapocalypse/core/api/BooksController.java]: Why You enable this auto configuration in the controller  ?

File: Core/src/main/java/javapocalypse/core/api/BooksController.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+@RequestMapping("/books")
+public class BooksController {
+
+    private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+
+    @Inject
+    public BooksController(BookRepository bookRepository, BookMapper bookMapper) {
```
Comment (egolesor) [inline Core/src/main/java/javapocalypse/core/api/BooksController.java:20]: :+1:

File: Core/src/main/java/javapocalypse/core/api/BooksController.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        this.bookRepository = bookRepository;
+        this.bookMapper = bookMapper;
+    }
+
+    @RequestMapping(path = "", method = RequestMethod.GET, produces = "application/json")
+    @ResponseBody
```
Comment (egolesor) [inline Core/src/main/java/javapocalypse/core/api/BooksController.java]: @RestController automatically add @ResponseBody at all methods so you may delete this.

File: Core/src/main/java/javapocalypse/core/api/BooksController.java (hunk 2 above)
Comment (egolesor) [reply Core/src/main/java/javapocalypse/core/api/BooksController.java:30]: @PostMapping ?

File: Core/src/main/java/javapocalypse/core/api/BooksController.java [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    public List<BookDTO> listBooks() {
+        return bookMapper.mapToDTOs(bookRepository.findAll());
+    }
+
+    @RequestMapping(path = "", method = RequestMethod.POST, consumes = "application/json", produces = "application/json")
+    @ResponseBody
```
Comment (egolesor) [inline Core/src/main/java/javapocalypse/core/api/BooksController.java]: Here again is no need for that

File: Core/src/main/java/javapocalypse/core/model/entities/Book.java [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+@Table(name = "books")
+@Data
+public class Book {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    @Column(name = "id")
+    private long id;
+
+    @Enumerated(EnumType.STRING)
```
Comment (egolesor) [inline Core/src/main/java/javapocalypse/core/model/entities/Book.java:21]: :+1:

File: Core/src/main/java/javapocalypse/core/model/entities/Book.java [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+    @Column(name = "new_price", nullable = false)
+    private String newPrice;
+
+    @Column(name = "tag", nullable = false)
+    private String tag;
+
+    public Book() {
+    }
+
+    public Book(
```
Comment (egolesor) [inline Core/src/main/java/javapocalypse/core/model/entities/Book.java:46]: For this 2 methods you can use lombok if you used it so far

File: Core/src/main/java/javapocalypse/core/model/entities/Library.java [hunk 11]
```diff
@@ -0,0 +1,23 @@
+package javapocalypse.core.model.entities;
+
+public enum Library {
+
+    EMPIK_COM {
+        @Override
+        public String toString() {
+            return "empik_com";
```
Comment (egolesor) [inline Core/src/main/java/javapocalypse/core/model/entities/Library.java:8]: first time I see such implementation :+1:

**Comment by egolesor** [comment]
After fixing the issues change the wiki and the script to run the application or inform me to do that.

File: Core/src/main/java/javapocalypse/core/api/BooksController.java (hunk 2 above)
Comment (tdudzik) [reply Core/src/main/java/javapocalypse/core/api/BooksController.java:30]: @egolesor  what do you mean?

File: Core/src/main/java/javapocalypse/core/model/entities/Book.java (hunk 10 above)
Comment (tdudzik) [reply Core/src/main/java/javapocalypse/core/model/entities/Book.java:46]: No I can't. Notice that the second constructor does not have `id` param.

**Comment by tdudzik** [comment]
Me and Łukasz fucked up my branch and I need to create another pull request after resolving issues with rebasing.

**Pull request closed without merging**